### ISSUE TITLE
HiDive - fails to download correct subtitle file or detect multiple subtitle files

### ISSUE THREAD

**Issue by orbitalflower** [ytdl-org/youtube-dl#23830]
## Checklist

- [x] I'm reporting a broken site support
- [x] I've verified that I'm running youtube-dl version **2020.01.24**
- [x] I've checked that all provided URLs are alive and playable in a browser
- [x] I've checked that all URLs and arguments with special characters are properly quoted or escaped
- [x] I've searched the bugtracker for similar issues including closed ones

## Verbose log

```
$ youtube-dl -v --write-sub --all-subs -f worst "https://www.hidive.com/stream/food-wars/s01e001"
[debug] System config: []
[debug] User config: [u'--no-call-home']
[debug] Custom config: []
[debug] Command-line args: [u'-v', u'--write-sub', u'--all-subs', u'-f', u'worst', u'https://www.hidive.com/stream/food-wars/s01e001']
[debug] Encodings: locale UTF-8, fs UTF-8, out UTF-8, pref UTF-8
[debug] youtube-dl version 2020.01.24
[debug] Python version 2.7.17 (CPython) - Linux-4.15.0-74-generic-x86_64-with-Ubuntu-18.04-bionic
[debug] exe versions: ffmpeg 3.4.6, ffprobe 3.4.6, phantomjs 2.1.1, rtmpdump 2.4
[debug] Proxy map: {}
[HiDive] food-wars/s01e001: Downloading JSON metadata
[HiDive] food-wars/s01e001: Downloading m3u8 information
[HiDive] food-wars/s01e001: Downloading m3u8 information
[HiDive] food-wars/s01e001: Downloading m3u8 information
[info] Writing video subtitles to: food-wars_s01e001-food-wars_s01e001.en.vtt
[debug] Invoking downloader on u'https://www.hidive.com/manifest/child/44e859ea5112f40a78407113c08a1dd9a0fd6096/FDW_s01e001_tv_or_na_ja_xx_HLS360p_16x9_00_v.m3u8'
[hlsnative] Downloading m3u8 manifest
[hlsnative] Total fragments: 151
[download] Destination: food-wars_s01e001-food-wars_s01e001.mp4
[download] 100% of 145.63MiB in 03:31
[debug] ffmpeg command line: ffprobe -show_streams 'file:food-wars_s01e001-food-wars_s01e001.mp4'
[ffmpeg] Fixing malformed AAC bitstream in "food-wars_s01e001-food-wars_s01e001.mp4"
[debug] ffmpeg command line: ffmpeg -y -loglevel 'repeat+info' -i 'file:food-wars_s01e001-food-wars_s01e001.mp4' -c copy -f mp4 '-bsf:a' aac_adtstoasc 'file:food-wars_s01e001-food-wars_s01e001.temp.mp4'
```

## Description

I'm trying to download a video from HiDive with the Japanese audio and English subtitles. However, it only downloads a limited subtitle file intended to accompany the English dub audio. This .vtt file contains Caption lines (for translating on-screen text and song lyrics) but not Subtitle lines (for translating spoken text).

Even using --all-subs, it only detects and downloads this Captions subtitle file. The full Subtitle file definitely exists, and appears in the browser requests. For comparison, it can be found at
<https://www.hidive.com/caption/vtt/d40c35ce045ef78d47fe727b3690ad27b4d1f8a9/FDW_s01e001_tv_hv_or_ja_en_v02.vtt>.

The output of --list-subs only shows one entry for English. If you look at the example URL, there are not only multiple English subtitles but other foreign-language versions which the youtube-dl extractor doesn't detect. Example:

```
$ youtube-dl --list-subs "https://www.hidive.com/stream/food-wars/s01e001"
[HiDive] food-wars/s01e001: Downloading JSON metadata
[HiDive] food-wars/s01e001: Downloading m3u8 information
[HiDive] food-wars/s01e001: Downloading m3u8 information
[HiDive] food-wars/s01e001: Downloading m3u8 information
Available subtitles for food-wars/s01e001:
Language formats
en       vtt, vtt, vtt
```

**Comment by Sophira** [comment]
I'm also having the same issues regarding downloading subtitle files, trying to get the *Kase-san and Morning Glories* OVA: https://www.hidive.com/stream/kase-san-and-morning-glories/2018060901 . It seems to download a subtitle file for a dubbed (English-language) version (my guess, because it doesn't include any dialogue but does include written text), and there's no way to get a subtitle file for the subbed (Japanese-language) version.

**Comment by darkhelmet2016** [comment]
> I'm also having the same issues regarding downloading subtitle files, trying to get the _Kase-san and Morning Glories_ OVA: https://www.hidive.com/stream/kase-san-and-morning-glories/2018060901 . It seems to download a subtitle file for a dubbed (English-language) version (my guess, because it doesn't include any dialogue but does include written text), and there's no way to get a subtitle file for the subbed (Japanese-language) version.

It contains 2 separate sub files, one is for on screen translation, one for the Japanese audio. The English dub is under Eng Caps but if you want the subs for the Japanese audio is under English subs if they need to fix the program to detect them.

**Comment by darkhelmet2016** [comment]
One thing should mention thought is that if you are looking to mux the file that you can only do mp4 video with srt subs and mkv with srt or ass which means you will need to add in a line to covert the subs from vtt.

**Comment by orbitalflower** [comment]
Note that there are a limited number of series that you can access without needing an account. The episode in my example, <https://www.hidive.com/stream/food-wars/s01e001>, is currently one episode which can be accessed without a login, and can be used to test this.

The page <https://www.hidive.com/free-episodes> will list others. Anything marked "dubbed" has multiple subtitle files and can therefore be used to test this bug.

**Comment by darkhelmet2016** [comment]
The extractor here still needs to be updated as it does not have lines in it to download subs like in Spanish as they had started subbing in other languages long after it was released and updated. It has no parameters to find all the subs that some shows now contain and some shows even have a broadcast dub as well as a home version dub and or sub that need to be detected as well.

**Comment by Stzam** [comment]
Hi i was wondering what is the url to download stuff from hidive    Youtube-dl -   hidive url every time i try it says i have to log in how to i log in what is the Youtube dl comand to log in and download or choose quality

**Comment by darkhelmet2016** [comment]
Think it would be youtube-dl --username --password url for the format unless they use cookies then think it would be like  youtube-dl --cookies "path for cookies file" url.

**Comment by Stzam** [comment]
Hi darkhelmet thanks for responding so you mean    youtube-dl --username --password https://www.hidive.com/stream/btooom/s01e001    i tried that and it dont work i tried youtube-dl --https://www.hidive.com/stream/the-fruit-of-grisaia/s01e001"  what command do  you use when your downloading from hidive ?

**Comment by darkhelmet2016** [comment]
You have to actually put in your user name and password, do you have a hidive account? It will be your email and your password for the site.

**Comment by Stzam** [comment]
yh i do have a hidive account i typed my email password and the link i type like this 
youtube-dl --email --password https://www.hidive.com/stream/btooom/s01e001  it says error no such option and then shows my email

**Comment by darkhelmet2016** [comment]
['--newline', '-u', 'PRIVATE', '-p', 'PRIVATE', '-i', '-o', 'C:\\Users\\Owner\\Downloads\\%(title)s-%(height)sp.%(ext)s', '-v', '--hls-prefer-native', '--add-metadata', 'https://www.hidive.com/stream/food-wars/s01e001'] Is what use for mine so it will write in show data. Where it says private you put in your email and password.

**Comment by Stzam** [comment]
Thanks darkhelmet could you dumb it down abit for me tho   so like this  ['--youtube-dl, '-u', 'Fakeemail@gmail.com', '-p', 'password', '-i', '-o', 'C:\Users\Owner\Downloads\%(1)s-%(1920x1080)sp.%(ext)s', '-v', '--hls-prefer-native', '--add-metadata', 'https://www.hidive.com/stream/btooom/s01e001']  sorry dude im  a idiot lol

**Comment by darkhelmet2016** [comment]
youtube-dl -u email address -p pasword https://www.hidive.com/stream/btooom/s01e001

**Comment by darkhelmet2016** [comment]
@ C:\Users\Owner\youtube-dl

> youtube-dl -u pivate -p private https://www.hidive.com/s
tream/btooom/s01e001
[HiDive] None: Downloading webpage
[HiDive] Logging in
[HiDive] btooom/s01e001: Downloading JSON metadata
[HiDive] btooom/s01e001: Downloading m3u8 information
[HiDive] btooom/s01e001: Downloading m3u8 information
[hlsnative] Downloading m3u8 manifest
[hlsnative] Total fragments: 148
[download] Destination: btooom_s01e001-btooom_s01e001.mp4
[download]  64.1% of ~551.50MiB at  4.14MiB/s ETA 01:42

**Comment by Stzam** [comment]
darkhelmet thank you so much your awesome  i really appericate it worked is there a way i can choose the quality cause i want best quality 1080p

**Comment by darkhelmet2016** [comment]
That will give you 1080p by default.

**Comment by Stzam** [comment]
Thanks dude your amazing

**Comment by darkhelmet2016** [comment]
NP

**Comment by Stzam** [comment]
darkhelmet dude sorry to be annoying but is there a way to get the sub with the download or do i need to download that separately

**Comment by darkhelmet2016** [comment]
As for the food wars problem youtube-dl needs to be able to grab the following subs, id="tv_hv_ja_en", id="tv_br_ja_en", id="tv_br_ja_pt", id="tv_br_ja_sp", id="tv_hv_en_xx", and id="tv_hv_en_en" in the extractor. Other shows may require additional detection.

**Comment by darkhelmet2016** [comment]
youtube-dl -u -p  --write-sub --all-subs https://www.hidive.com/stream/btooom/s01e001 Should do it but not sure if they have fixed the detection issue with the proper subs.

**Comment by Stzam** [comment]
Didn't work dude they must have no fixed it yet thanks tho dude for all the help  much appreciated

**Comment by darkhelmet2016** [comment]
Its ok, but yea they need a lot of updating on this one since there are multiple streams for subs as well as video since they have shows in English, Japanese and even Spanish now for the audio language as well as home versions and broadcast versions too.

**Comment by Stzam** [comment]
yh hopefully they will sort it out i originally just tried using video download helper to try get the links but there drm protected which is weird because i don't think  Vrv crunchyroll or funimation are  this way you get subs and english versions

**Comment by Stzam** [comment]
darkhelmet  hidive has uncensored versions of shows the one i downloaded was uncensored but the one the site was is there a command where i can choose version like the  720p 1080p or a command to list versions of the file

**Comment by darkhelmet2016** [comment]
-F, --list-formats List all available formats of requested videos. -f "best[height=720]" -f "best[height=1080]"

**Comment by Stzam** [comment]
Thanks for responding dark helmet so you mean like this youtube-dl -u -p  https://www.hidive.com/stream/btooom/s01e001 -F, ---F, --list-formats    and it will show me could you help me with the right command im a idiot

**Comment by darkhelmet2016** [comment]
youtube-dl -F -u -p https://www.hidive.com/stream/btooom/s01e001

**Comment by darkhelmet2016** [comment]
-F is the short version and --list-formats is the long version of the same command.

**Comment by Stzam** [comment]
thanks darkhelmet your amazing it show me the format how do i chose one like what command i type im so stupid

**Comment by darkhelmet2016** [comment]
youtube-dl  -f "best[height=720]" -u -p https://www.hidive.com/stream/btooom/s01e001 will give you 720p and youtube-dl  -f "best[height=1080]" -u -p https://www.hidive.com/stream/btooom/s01e001 will give you 1080p to download.

**Comment by Stzam** [comment]
thanks again dude your awesome

**Comment by darkhelmet2016** [comment]
The problem with the extractor for hidive is it needs to know if the video is home video hv or broadcast br and then needs to detect if the subs are English en, Japanese jp, Portuguese pt or Spanish LatAm sp. The https://www.hidive.com/tv page will give you a list of formats for audio and subtitles that will need supporting. It will need sub support for English, Latin American Spanish, Portuguese, Arabic, German, European Spanish, French, Italian, and Russian subs.

**Comment by darkhelmet2016** [comment]
But then on the movies page it also has Korean listed for audio. https://www.hidive.com/movies

**Comment by Stzam** [comment]
darkhelmet is there a similar command line for vrv.co if i want to download from there with youtubedl

**Comment by darkhelmet2016** [comment]
It should be the same as the hidive one unless you need subs. But just a warning their subs and subs use different links for the same show unlike hidive. You will need to go to the show page and chose which version you want of the show to get the url from.

**Comment by Stzam** [comment]
Hi darkhelmet i tried this command 
youtube-dl -u -p pasword  https://vrv.co/watch/GREX7JW8Y/Mob-Psycho-100:Ripped-Apart-Someone-Is-Watching  and nothing is happening is my command wrong or maybe it just don't support vrv anymore

**Comment by darkhelmet2016** [comment]
It still works
@ C:\Users\Owner\youtube-dl

> youtube-dl -u  -p https://vrv.co/watch/GREX7JW8
Y/Mob-Psycho-100:Ripped-Apart-Someone-Is-Watching
[vrv] None: Downloading webpage
[vrv] Downloading Token Credentials JSON metadata
[vrv] GREX7JW8Y: Downloading resource path JSON metadata
[vrv] GREX7JW8Y: Downloading CMS Signing JSON metadata
[vrv] GREX7JW8Y: Downloading object JSON metadata
[vrv] GREX7JW8Y: Downloading video JSON metadata
[vrv] GREX7JW8Y: Downloading streams JSON metadata
[vrv] GREX7JW8Y: Downloading hls-audio-ja-JP information
[vrv] GREX7JW8Y: Downloading hls-audio-ja-JP-hardsub-ar-ME information
[vrv] GREX7JW8Y: Downloading hls-audio-ja-JP-hardsub-fr-FR information
[vrv] GREX7JW8Y: Downloading hls-audio-ja-JP-hardsub-ru-RU information
[vrv] GREX7JW8Y: Downloading hls-audio-ja-JP-hardsub-de-DE information
[vrv] GREX7JW8Y: Downloading hls-audio-ja-JP-hardsub-es-LA information
[vrv] GREX7JW8Y: Downloading hls-audio-ja-JP-hardsub-pt-BR information
[vrv] GREX7JW8Y: Downloading hls-audio-ja-JP-hardsub-es-ES information
[vrv] GREX7JW8Y: Downloading hls-audio-ja-JP-hardsub-en-US information
[vrv] GREX7JW8Y: Downloading hls-audio-ja-JP-hardsub-it-IT information
[vrv] GREX7JW8Y: Downloading dash-audio-ja-JP information
[vrv] GREX7JW8Y: Downloading dash-audio-ja-JP-hardsub-ar-ME information
[vrv] GREX7JW8Y: Downloading dash-audio-ja-JP-hardsub-fr-FR information
[vrv] GREX7JW8Y: Downloading dash-audio-ja-JP-hardsub-ru-RU information
[vrv] GREX7JW8Y: Downloading dash-audio-ja-JP-hardsub-de-DE information
[vrv] GREX7JW8Y: Downloading dash-audio-ja-JP-hardsub-es-LA information
[vrv] GREX7JW8Y: Downloading dash-audio-ja-JP-hardsub-pt-BR information
[vrv] GREX7JW8Y: Downloading dash-audio-ja-JP-hardsub-es-ES information
[vrv] GREX7JW8Y: Downloading dash-audio-ja-JP-hardsub-en-US information
[vrv] GREX7JW8Y: Downloading dash-audio-ja-JP-hardsub-it-IT information
[dashsegments] Total fragments: 148
[download] Destination: Ripped Apart ~Someone Is Watching~-GREX7JW8Y.fdash-audio
-ja-JP-hardsub-de-DE-f2-v1-x3.mp4
[download]   9.8% of ~1.10GiB at  4.22MiB/s ETA 03:17

**Comment by Stzam** [comment]
yh it works  but the video has no audio when it is downloaded its weird it just captures video

**Comment by Stzam** [comment]
dark helmet i found the proBLEMwhen i list formats there are links without audio  i think this will work but i don't know how to register the format what do i put here for fomat bit   youtube-dl --format EMPTY -u -p  https://vrv.co/watch/GREX7JW8

**Comment by Stzam** [comment]
hls-audio-ja-JP-3078-2     mp4        1920x1080  [ja-JP] 3078k , avc1.640028, 23
.974fps, mp4a.40.2 (best)   this is the best format but which part of it to i put for youtube-dl to register it

**Comment by darkhelmet2016** [comment]
What is the line you are using because it should download both the video and audio with the line gave you.

**Comment by Stzam** [comment]
im using this command  - youtube-dl -u -p --write-sub --all-subs https://vrv.co/watch/GREX7JW8        it is downloading the files separate the audio the video and the sub  i get the sub being a separate file but shouldn't the video and audio be one file 
![Screenshot (2720)](https://user-images.githubusercontent.com/61325852/75250809-1b898480-57d1-11ea-9602-bc3eb1195c44.png)

**Comment by darkhelmet2016** [comment]
Not until the file is finished and it should give you a message at the end assuming you are using a command prompt for this.
[ffmpeg] Merging formats into "Ripped Apart ~Someone Is Watching~-GREX7JW8Y.mp4"

Deleting original file Ripped Apart ~Someone Is Watching~-GREX7JW8Y.fdash-audio-
ja-JP-hardsub-de-DE-f2-v1-x3.mp4 (pass -k to keep)
Deleting original file Ripped Apart ~Someone Is Watching~-GREX7JW8Y.fdash-audio-
ja-JP-hardsub-ru-RU-f1-a1-x3.m4a (pass -k to keep)

**Comment by Stzam** [comment]
this happens to me  [dashsegments] Total fragments: 148
[download] Destination: Ripped Apart ~Someone Is Watching~-GREX7JW8.fdash-audio-
ja-JP-hardsub-de-DE-f2-v1-x3.mp4
[download] 100% of 1.35GiB in 01:54:48
[dashsegments] Total fragments: 148
[download] Destination: Ripped Apart ~Someone Is Watching~-GREX7JW8.fdash-audio-
ja-JP-hardsub-ru-RU-f1-a1-x3.m4a
[download] 100% of 22.63MiB in 03:19
[ffmpeg] Merging formats into "Ripped Apart ~Someone Is Watching~-GREX7JW8.mp4"
ERROR: file:Ripped Apart ~Someone Is Watching~-GREX7JW8.fdash-audio-ja-JP-hardsu
b-de-DE-f2-v1-x3.mp4: No such file or directory

**Comment by darkhelmet2016** [comment]
Might not be a good idea to have this running in that folder. If possible try running it from the desktop in a folder just for this program but make sure you also have all the files for ffmpeg from the source where you downloaded it from.
